main_loop_tf/validate.py

```diff
@@ -12,16 +12,22 @@
 from tqdm import tqdm
 import tensorflow as tf
 
-from utils import compute_chunk_size, fig2array
+from utils import compute_chunks, fig2array
 
 
 def validate(placeholders,
              eval_outs,
              val_summary_op,
              val_reset_cm_op,
+             smaller_n_gpus,
              which_set='valid',
              epoch_id=None):
 
+    val_inputs_per_gpu, labels_per_gpu = placeholders
```

Comment:
Loose these variables. You use them in the next 2 lines only, and the names are longer than placeholders[i]...